app/controllers/tags_controller.rb

```diff
@@ -129,25 +129,32 @@ def merge
     AuditLog.moderator_audit event_type: 'tag_merge', related: @primary, user: current_user,
```

Comment:
Would it be useful to make this log entry after the merge, so that it only gets recorded when successful? I don't know to what extent Rails already handles logging what was being attempted if there is an error. Do we need to check for and log errors ourselves?

Reply:
Good remark (though not code I touched :P). Really, all of this should be wrapped in a transaction. That means it either all applies, or none of it applies, to avoid exactly the case where we get partial execution. I will commit that.